accepted/global-json-updates.md

```diff
@@ -0,0 +1,315 @@
+  * The impact of this is mitigated by work we did in .NET Core 3.0 Preview 3 to replace earlier patch versions.
+* Due to #3, users can not guarantee they are pinned to a specific version.
+
+## Proposal
+
+### Roll-forward
+
+.NET Core SDK "roll-forward" is problematic because version numbers are not linear. For example: 2.1.600 contains MSBuild 16 while 2.2.100 contains MSBuild 15.9. This is not a new problem, since it already occurs when there is no `global.json`. However, we do not want to roll-forward in scenarios that might surprise the user. Because of this, roll-forward will be allowed only in limited scenarios.
+
+### Details
+
+The `global.json` schema to offers support for all of the identified scenarios that can be fixed without altering how we find `global.json`:
+
+* `sdk`
+  * `version`
+    * The lowest SDK the resolver would select - any SDKs with lower versions are ignored.
+    * Wildcards are not supported.
+    * Default: empty
+  * `ignorePreview`
+    * `true` or `false`
+    * Default: `false` to match previous behavior.
+    * Alternate names: `releaseOnly` or `useReleaseOnly`
+    * This is new behavior.
+  * `rollForward`
+    * Lower version numbers are always ignored. 
+    * `patch`
+      * If the requested major, minor, feature band, and patch is found, use it.
+      * Otherwise, use the highest patch within the specified major, minor and feature band.
+      * This policy means the latest patches, including security patches will not be used unless the specified version is remove from the machine.
+      * This matches previous behavior when a `global.json` file contains a version.
+    * `latestPatch`
+      * Even if the requested major, minor, feature band and patch is found, do not use it unless it matches rules below.
+      * Select the highest available patch version that matches the specified major, minor and feature band versions.
+      * This is new behavior.
+    * `disable`
+      * Do not roll forward. Only bind to specified version. 
+      * This policy means the latest patches, including security patches will not be used.
+      * This is new behavior.
+    * `latest`
+      * Even if the requested major, minor, feature band and patch is found, do not use it unless it matches rules below.
+      * Select the highest available major version, and select it's highest minor version, and that minor's highest feature band, and that feature band's highest patch.
+      * Using latest with a version is new behavior. Otherwise this matches previous behavior when there is no `global.json` or no version is specified.
+    * Defaults: 
+      * Defaults are defined to support backward compatibility issues.
+      * If a version is specified in the `global.json`, the default is `patch` to match previous behavior.
+      * If there is no `global.json` or a version is not specified, the default is`latest`, again to match previous behavior.
+
```

Comment:
latest is a big hammer and patch, latest patch, and disable are too rigid in the face of upgrades/sdk clean up. I think we should mirror the minor and major options of the runtime. They are conservative and do not go to latest always. 

I also don't think we should be too concerned about using latestPatch by default even if exact is found. Patches are always highly compatible and contain security fixes. You should have to opt out of them instead of opting in to them.

My suggestion would be to have the same rollForward options as the runtime, but consider version Major.Minor.(Band * 100) + Patch. To be Major.(Minor * 100 + Band).Patch.

I would further suggest changing the default rollForward to Major. I don't consider this a significant breaking change. You will get security patches by default, and otherwise turn error cases into finding nearest match.

Since we will be stop leaving SDKs behind in VS, this will be a much more compatible default. Only users that explicitly opt in to more conservative roll forward will get hard broken by upgrades or sdk clean up.